src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -831,64 +840,152 @@ window.qBittorrent.DynamicTable ??= (() => {
-                        if (rowPos === j)
+                for (let rowPos = 0; rowPos < rows.length; ++rowPos) {
+                    const rowId = rows[rowPos]["rowId"];
+                    let tr_found = false;
+                    for (let j = rowPos; j < trs.length; ++j) {
+                        if (trs[j]["rowId"] === rowId) {
+                            tr_found = true;
+                            if (rowPos === j)
+                                break;
+                            trs[j].inject(trs[rowPos], "before");
+                            const tmpTr = trs[j];
+                            trs.splice(j, 1);
+                            trs.splice(rowPos, 0, tmpTr);
                             break;
-                        trs[j].inject(trs[rowPos], "before");
-                        const tmpTr = trs[j];
-                        trs.splice(j, 1);
-                        trs.splice(rowPos, 0, tmpTr);
-                        break;
+                        }
+                    }
+                    if (tr_found) { // row already exists in the table
+                        this.updateRow(trs[rowPos], fullUpdate);
+                    }
+                    else { // else create a new row in the table
+                        const tr = this.createRowElement(rows[rowPos]);
+
+                        // Insert
+                        if (rowPos >= trs.length) {
+                            tr.inject(this.tableBody);
+                            trs.push(tr);
+                        }
+                        else {
+                            tr.inject(trs[rowPos], "before");
+                            trs.splice(rowPos, 0, tr);
+                        }
+
+                        // Update context menu
+                        this.contextMenu?.addTarget(tr);
+
+                        this.updateRow(tr, true);
                     }
                 }
-                if (tr_found) { // row already exists in the table
-                    this.updateRow(trs[rowPos], fullUpdate);
+
+                const rowPos = rows.length;
+
+                while ((rowPos < trs.length) && (trs.length > 0))
+                    trs.pop().destroy();
+            }
+        },
+
+        rerender: function() {
+            if (!this.useVirtualList)
+                return;
+            const rows = this.getFilteredAndSortedRows();
+
+            for (let i = 0; i < this.selectedRows.length; ++i) {
+                if (!(this.selectedRows[i] in rows)) {
+                    this.selectedRows.splice(i, 1);
+                    --i;
                 }
-                else { // else create a new row in the table
-                    const tr = document.createElement("tr");
-                    // set tabindex so element receives keydown events
-                    // more info: https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event
-                    tr.tabIndex = -1;
+            }
```

Comment:
I suppose you should use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter instead.